topics/tutorials/file_and_code_templates/providing_file_templates.md

```diff
@@ -0,0 +1,97 @@
+It should contain only templates for given language core entities, e.g., a class, an interface, or other items that users very frequently create.
+If the language defines a lot of core entities, see [](using_file_templates.md#custom-create-file-from-template-actions) for a more user-friendly solution.
+
+#### Includes
+
+The _Includes_ category contains reusable fragments used for including in other file templates using the Apache Velocity `#parse()` directive, e.g., license header or documentation comment skeleton.
+It includes templates located in the <path>fileTemplates/includes</path> directory.
+
+#### Code
+
+The _Code_ category contains templates used for inserting in existing files, e.g., a code fragment used in an [intention action](code_intentions.md) or [quick fix](code_inspections.md#quick-fix-implementation).
+It includes templates located in the <path>fileTemplates/code</path> directory.
+
+#### Internal
+
+The _Internal_ category contains templates that are not visible in the IDE settings by default and cannot be edited by users.
+It includes templates located in the <path>fileTemplates/internal</path> directory.
+
+#### Other
+
+The _Other_ category contains other templates organized in groups.
+It includes templates located in the <path>fileTemplates/j2ee</path> directory and registered via the `com.intellij.fileTemplateGroup` extension point (EP).
+Note that the <path>j2ee</path> directory name is historical and unrelated to the J2EE technology.
+This category is intended for templates that are not used for creating core language entities or are used less frequently by users, e.g., a specific XML configuration file, a framework-specific class in Java language, etc.
+To include file templates in the _Other_ section of the <menupath>Settings/Preferences | Editor | File and Code Templates</menupath> settings page, provide an implementation of the
+[`FileTemplateGroupDescriptorFactory`](upsource:///platform/lang-api/src/com/intellij/ide/fileTemplates/FileTemplateGroupDescriptorFactory.java)
+and register it via the `com.intellij.fileTemplateGroup` EP.
+
+**Example**:
+[`MavenFileTemplateGroupFactory`](upsource:///plugins/maven/src/main/java/org/jetbrains/idea/maven/utils/MavenFileTemplateGroupFactory.java)
+
+Note that
+[`FileTemplateGroupDescriptor`](upsource:///platform/lang-api/src/com/intellij/ide/fileTemplates/FileTemplateGroupDescriptor.java)
+is a subclass of
+[`FileTemplateDescriptor`](upsource:///platform/core-api/src/com/intellij/ide/fileTemplates/FileTemplateDescriptor.java)`,
+which allows creating nested groups.
+
+## Creating File Template Description
+
+By default, the description contains generic text about the syntax and properties of the given template.
+It is highly recommended to provide a custom description explaining its purpose and any available properties.
+Overriding the default description is achieved by creating an HTML file with the name matching template's name but with the `.html` extension.
```

Comment:
`<path>`

Reply:
Are you sure about this one? File extensions are written this way even in ["SDK Docs Style Guide"](https://plugins.jetbrains.com/docs/intellij/sdk-style.html).
Please confirm I should change it. I will also align the style guide too if so.

Reply:
Yes, using `<path>` is correct.

Reply:
It's clarified by Patrick already, fixed.